src/ImmediateContext.cpp

```diff
@@ -4054,15 +4054,37 @@ bool ImmediateContext::ResourceAllocationFallback(ResourceAllocationContext thre
             }
         }
     };
+
+    //If index == 0 we are checking for object deletion, else subobject deletion
+    auto WasMemoryFreed = [&](int index) -> bool {
+        
+        UINT64 newSyncPoint[(UINT)COMMAND_LIST_TYPE::MAX_VALID];
+        auto DeletionManagerLocked = m_DeferredDeletionQueueManager.GetLocked();
+        bool newSyncPointExists = false;
+        if (index == 0) {
+            newSyncPointExists = DeletionManagerLocked->GetFenceValuesForObjectDeletion(newSyncPoint);

```

Comment:
This doesn't indicate that memory was freed, does it? This just indicates that the wait was done. We only will free memory here as part of the wait if the wait required flushing the current command list.

I think the right fix here is to plumb an output parameter from `WaitForSyncPoint` that indicates either that we flushed (which implies that `TrimDeletedObjects` was called), or if we really want, to make that output parameter actually be the value from the call to `TrimDeletedObjects` directly.

Reply:
My reasoning was that TrimDeletedObjects is the only place that we [pop()](https://github.com/microsoft/D3D12TranslationLayer/blob/master/src/ImmediateContext.cpp#L455) from the deletion queues, and we only pop if an object was destroyed and TrimDeletedObjects would return true. If we don't want to go this route, I can take one of the options that you mentioned. Maybe just the return true if we flushed since that's a bit less messy.

Reply:
Oh, yeah I'm following now. Ok, this seems fine. Maybe just a comment then to indicate that a change in sync point means that the objects we were waiting for have already been deleted.